### DISCUSSION TITLE
ITEM_PROGRESS with definable MIN and MAX value

### DISCUSSION THREAD

**Discussion by rozrabiak** [forntoh/LcdMenu#163] [Ideas]
Hi,
I'm trying the ITEM_PROGRESS. Default value is incremented from 0 to 1000.
But there is difficult to set range from 1 to 59 (minutes) using mapping.
It would be very useful when programmer can  set MIN and MAX (and step of increment) value when ITEM_PROGRESS is defined, for every ITEM_PROGRESS definition. Because many users need to set range. For example:
Range for hours is: 1 to 23 OR 1 to 12.
For percents is 1 to 100.
Minutes & seconds 0...59..

etc..

**Comment by rozrabiak** [comment]
example from my project:
mapping values from  ITEM_PROGRESS where values are from 0 to 1000L to 0..100%. That's work's ok.
But...
I writing above percents to eeprom as settings (as byte).

In setup (when program starting) I'm reading from eeprom value, for eg. 45 (as 45%).

When I put 45 to  ITEM_PROGRESS as setProgress(45) it's not a real value of 45.. Because progress are from 0 to 1000L as default...

Maybe anyone have any idea how to do it right?

Thank's

**Reply by forntoh** [reply]
Here you go #228 ☺️
